### PR TITLE
Fix python-augeas failure on Windows (v2)

### PR DESCRIPTION
**Pull request by adferrand**
This PR is a new approach for fixing #8732 based on the discussions occurred in the first PR #8877.

This PR upgrades python-augeas to the latest version, and avoids tests failure of Windows because of this upgrade. To do so it leverages the [tox multi-platform feature](https://tox.readthedocs.io/en/latest/example/platform.html) and modifications to `tools/venv.py` in order to not install and not test `certbot-apache` on Windows.

### CODE REVIEW

**Review by bmw — COMMENTED** [review]
I think there are a few problems here:

1. I think the `platform` value should be `win: win32` because we currently have errors like [this](https://dev.azure.com/certbot/certbot/_build/results?buildId=4317&view=logs&j=ab8d0e2b-dcbf-5ba2-5207-204c973e249b&t=9264c6c5-ee93-5b02-d751-bee067dd5ea0&l=234). In case the logs expire, text is `SKIPPED:  py38-cover-win: platform mismatch ('win32' does not match 'windows')`
2. `{lint,mypy}-{win,posix}` don't match the lint and mypy environments and instead run our unit tests.
3. I think we may want to add a platform line like `posix: ^(?!.*win32).*$`. Otherwise I think `tox` on Windows will try to run tests for both platforms.

**Comment by adferrand** [comment]
For 1. and 3., indeed I missed the targeted behavior because my version of code on Windows was not synced with the PR itself. In fact it was using the default platform name `linux` and `win` as provided by tox by default and proposed by the official documentation. Whoever I do not know if the `linux` default platform also matches Mac OS X.

Anyway I defined the `posix` platform with the regex you suggested.

**Comment by adferrand** [comment]
For 2., here is my proposition that replaces `lint` and `mypy` envs in favor of `lint-{win,posix}` and `mypy-{win,posix}`.

**Review by bmw — COMMENTED** [review]
Other than my inline comments, can you confirm `tox -e lint-win,mypy-win` are passing for you on Windows? They weren't for me on a machine I quickly spun up, but I think the problem was unrelated to this PR.

I'm curious because I notice we're still passing `certbot-apache` to both `pylint` and `mypy` and I wasn't sure if that'd cause problems.

**Comment by adferrand** [comment]
Yes, I confirm that `tox -e lint-win,mypy-win` is working as expected on Windows and before I start to study the propositions made in the review to improve the definition of the tox environments.

**Comment by adferrand** [comment]
(and passing `certbot-apache` in sources is not causing problem, though both pylint and mypy are complaining on unrelated errors specifically on Windows.)

File: tox.ini [hunk 1]
```diff
@@ -112,20 +117,22 @@ commands =
     python {toxinidir}/tools/pip_install.py mock
     {[base]install_and_test} {[base]all_packages}
 
-[testenv:lint]
+[testenv:lint-{win,posix}]
```
[the remark this answers is not in the document]
Comment (adferrand) [reply tox.ini:120]: Thanks! I was about to test that change to see if we can still keep the `lint` and `mypy` targets as before. It seems to work as expected.

**Review by bmw — APPROVED** [review]

**Pull request merged**